template-parts/content-grid.php

```diff
@@ -3,18 +3,43 @@
  * Template part for displaying grid posts.
  *
  * @package siteorigin-unwind
- * @since siteorigin-unwind 1.1 
+ * @since siteorigin-unwind 1.1
  * @license GPL 2.0
  */
+
+$gallery = get_post_gallery( get_the_ID(), false );
+if ( ! empty( $gallery ) && ! has_action( 'wp_footer', 'siteorigin_unwind_enqueue_flexslider' ) ) {
+	add_action( 'wp_footer', 'siteorigin_unwind_enqueue_flexslider' );
+}
 ?>
 
 <article id="post-<?php the_ID(); ?>" <?php post_class( 'archive-entry' ); ?>>
 
-	<?php if ( has_post_thumbnail() && siteorigin_setting( 'blog_featured_archive' ) ) : ?>
+	<?php if ( siteorigin_setting( 'blog_featured_archive' ) ) : ?>
 		<div class="entry-thumbnail">
-			<a href="<?php the_permalink() ?>">
-				<?php the_post_thumbnail( 'siteorigin-unwind-360x238-crop' ); ?>
-			</a>
+			<?php if ( get_post_format() == 'gallery' && $gallery != '' ) : ?>
+				<div class="flexslider gallery-format-slider">
+					<ul class="slides gallery-format-slides">
+						<?php foreach ( $gallery['src'] as $image ) : ?>
+							<li class="gallery-format-slide">
+								<img src="<?php echo $image; ?>">
+							</li>
+						<?php endforeach; ?>
+					<ul>
+				</div>
+			<?php elseif ( get_post_format() == 'image' && siteorigin_unwind_get_image() ) : ?>
+				<div class="entry-image">
+					<?php echo siteorigin_unwind_get_image(); ?>
```

Comment:
I think we should link the image to the post, basically, treat it like a featured image.